azurelinuxagent/common/osutil/clearlinux.py

```diff
@@ -79,14 +80,14 @@ def del_root_password(self):
                 passwd_content = fileutil.read_file(passwd_file_path)
                 if not passwd_content:
                     # Empty file is no better than no file
-                    raise FileNotFoundError # pylint: disable=undefined-variable
-            except FileNotFoundError: # pylint: disable=undefined-variable
+                    raise IOError
+            except (IOError, OSError):
```

Comment:
why OSError?

Reply:
I can look up and link the specifics, but it has to do with python2/3 compatibility.

Reply:
Found the [stackoverflow post I referenced](https://stackoverflow.com/questions/28633555/how-to-handle-filenotfounderror-when-try-except-ioerror-does-not-catch-it).

Basically, `FileNotFoundError` is a subclass of `OSError` and a python3 exclusive, whereas `IOError` (again, in python3) is just an alias for `OSError`; in python2, though, file operations throw `IOError`s (though, again, **not** `FileNotFoundError`s). `IOError` is therefore needed to catch in python2, and `OSError` is the correct error to catch for python3.

I suppose we could just use `IOError` (as it is the error thrown in python2 and an alias for the error thrown in python3), but I think it's better to be more explicit with the `OSError` we're actually catching in python3.

Reply:
OK